### PR TITLE
Pin debootstrap to tag 1.0.134 and fix azurelinux 3.0 build

### PR DESCRIPTION
**Pull request by am11**
While https://github.com/microsoft/azurelinux/pull/9422 is under review, we can pin it to ubuntu1 release tag.

We also need to add some more dependencies since azurelinux 3.0 rebuilt their packages and now doesn't install some transitive dependencies anymore.

### CODE REVIEW

**Comment by am11** [comment]
cc @akoeplinger @sbomer once upstream PR is merged and package is published, we can delete this entire `RUN` step.

File: src/azurelinux/3.0/net9.0/cross/riscv64/Dockerfile [hunk 1]
```diff
@@ -3,8 +3,7 @@ ARG ROOTFS_DIR=/crossrootfs/riscv64
 
 # Install the latest debootstrap
 RUN tdnf remove -y debootstrap && \
-    git clone --depth 1 "https://git.launchpad.net/ubuntu/+source/debootstrap" /debootstrap && \
-    chmod a+x /debootstrap/debootstrap && \
+    curl -sSL https://deb.debian.org/debian/pool/main/d/debootstrap/debootstrap_1.0.134.tar.gz | tar xzf - -C / && \
```
Comment (akoeplinger) [inline src/azurelinux/3.0/net9.0/cross/riscv64/Dockerfile:6]: would you mind adding sha checksum validation like we do e.g. here:
https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/0ca583b11fc3fc618335bf9394d79daba6884739/src/azurelinux/3.0/net9.0/cross/armv6/Dockerfile#L4-L10

**Review by akoeplinger — APPROVED** [review]

**Review by akoeplinger — APPROVED** [review]

**Comment by am11** [comment]
Looks like we hit an unrelated networking issue:

```
#5 [2/2] RUN tdnf install -y         awk         build-essential         ca-certificates         git         icu         rpm-build         ruby         shadow-utils         tar         util-linux         sudo     && tdnf clean all     && gem install fpm
#5 0.262 Loaded plugin: tdnfrepogpgcheck
#5 0.271 Refreshing metadata for: 'Azure Linux Official Preview 3.0 x86_64'
#5 0.684 Refreshing metadata for: 'Azure Linux Official Microsoft Non-Open-Source Preview 3.0 x86_64'
#5 0.798 awk package not found or not installed
#5 0.798 Error(1011) : No matching packages
```

**Comment by am11** [comment]
Looks like azure linux is having a down time. From local:

```sh
$ docker run --rm azurelinuxpreview.azurecr.io/public/azurelinux/base/core:3.0 tdnf install -y git  
Loaded plugin: tdnfrepogpgcheck
Refreshing metadata for: 'Azure Linux Official Preview 3.0 aarch64'
Refreshing metadata for: 'Azure Linux Official Microsoft Non-Open-Source Preview 3.0 aarch64'
git package not found or not installed
Error(1011) : No matching packages
```

**Comment by akoeplinger** [comment]
It seems it doesn't affect all packages, e.g. wget can still be installed. I sent an email to an internal mailing list I found.

**Comment by am11** [comment]
Fingers crossed. Looks like mostly dotnet org is using their docker image in CI https://github.com/search?q=azurelinuxpreview.azurecr.io&type=code. Based on these stats, priority could be low.

**Comment by mthalman** [comment]
Yeah, they're repopulating their packages. So we just need to wait.

**Comment by akoeplinger** [comment]
I got confirmation that the repo rebuild is done now, but we're still seeing a failure:
>  CMake Error: CMake was unable to find a build program corresponding to "Unix Makefiles".

It looks like `make` is no longer installed as a dependency by the packages we install in https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/69649a0c41a0e2c96eb8979ddb63b78ff76fe536/src/azurelinux/3.0/net9.0/crossdeps-builder/amd64/Dockerfile#L3

We can probably just add it but now I'm wondering what other packages we're missing.

**Comment by akoeplinger** [comment]
I was able to grab the packages which were installed from an earlier CI log and diffed it with the current one: https://gist.github.com/akoeplinger/c98960652058673a3ee8734660be6802

I don't think we need any of those except make?

**Comment by am11** [comment]
Cherry-picked your commits here for green CI (for some reason, they both need each other now 🤷‍♀️).

**Comment by am11** [comment]
Looks like we also need `npm` now that `nodejs` doesn't transitively pull it.

**Comment by am11** [comment]
@akoeplinger green build! 🙏 😅

File: src/cbl-mariner/2.0/webassembly/amd64/Dockerfile [hunk 2]
```diff
@@ -6,6 +6,7 @@ RUN tdnf update -y \
         # WebAssembly build dependencies
         which \
         nodejs \
+        npm \
```
Comment (akoeplinger) [inline src/cbl-mariner/2.0/webassembly/amd64/Dockerfile:9]: we shouldn't need to change the mariner 2.0 images, only 3.0 was affected afaik

File: src/azurelinux/3.0/net9.0/webassembly/amd64/Dockerfile [hunk 3]
```diff
@@ -6,6 +6,7 @@ RUN tdnf update -y \
         # WebAssembly build dependencies
         which \
         nodejs \
+        npm \
```
Comment (akoeplinger) [inline src/azurelinux/3.0/net9.0/webassembly/amd64/Dockerfile:9]: probably should add it to the net8 image too?

File: src/cbl-mariner/2.0/webassembly/amd64/Dockerfile (hunk 2 above)
Comment (am11) [reply src/cbl-mariner/2.0/webassembly/amd64/Dockerfile:9]: I think there is a confusion. nodejs used to install npm transitively, now it doesn't. The next time this image will be built we will be missing npm. The failing build didn't built cbl images. I updated it "while I was at it" so the next person don't run into it.

File: src/azurelinux/3.0/net9.0/webassembly/amd64/Dockerfile (hunk 3 above)
Comment (am11) [reply src/azurelinux/3.0/net9.0/webassembly/amd64/Dockerfile:9]: Ah I was searching with `nodejs \`, didn't realized in net8.0 files it was the last one in the list 😅

File: src/cbl-mariner/2.0/webassembly/amd64/Dockerfile (hunk 2 above)
Comment (akoeplinger) [reply src/cbl-mariner/2.0/webassembly/amd64/Dockerfile:9]: that's the case for 3.0, but in 2.0 the nodejs package still provides the npm command:

```
$ docker run --rm mcr.microsoft.com/cbl-mariner/base/core:2.0 sh -c "tdnf install -y nodejs && npm --help"
Loaded plugin: tdnfrepogpgcheck
Refreshing metadata for: 'CBL-Mariner Official Microsoft 2.0 aarch64'
Refreshing metadata for: 'CBL-Mariner Official Extras 2.0 aarch64'
Refreshing metadata for: 'CBL-Mariner Official Base 2.0 aarch64'

Installing:
c-ares           aarch64      1.19.1-1.cm2     mariner-official-base 278.56k   121.55k
brotli           aarch64      1.0.9-2.cm2      mariner-official-base   1.48M   738.43k
nodejs18         aarch64      18.20.2-2.cm2    mariner-official-base  54.95M    16.43M

Total installed size:  56.70M
Total download size:  17.27M

importing key from file:///etc/pki/rpm-gpg/MICROSOFT-RPM-GPG-KEY
importing key from file:///etc/pki/rpm-gpg/MICROSOFT-METADATA-GPG-KEY


Testing transaction
Running transaction
Installing/Updating: brotli-1.0.9-2.cm2.aarch64
Installing/Updating: c-ares-1.19.1-1.cm2.aarch64
Installing/Updating: nodejs18-18.20.2-2.cm2.aarch64
npm <command>

Usage:

npm install        install all the dependencies in your project
npm install <foo>  add the <foo> dependency to your project
npm test           run this project's tests
npm run <foo>      run the script named <foo>
npm <command> -h   quick help on <command>
npm -l             display usage info for all commands
npm help <term>    search for help on <term>
npm help npm       more involved overview

All commands:

    access, adduser, audit, bugs, cache, ci, completion,
    config, dedupe, deprecate, diff, dist-tag, docs, doctor,
    edit, exec, explain, explore, find-dupes, fund, get, help,
    help-search, hook, init, install, install-ci-test,
    install-test, link, ll, login, logout, ls, org, outdated,
    owner, pack, ping, pkg, prefix, profile, prune, publish,
    query, rebuild, repo, restart, root, run-script, sbom,
    search, set, shrinkwrap, star, stars, start, stop, team,
    test, token, uninstall, unpublish, unstar, update, version,
    view, whoami

Specify configs in the ini-formatted file:
    /root/.npmrc
or on the command line via: npm <command> --key=value

More configuration info: npm help config
Configuration fields: npm help 7 config

npm@10.5.0 /usr/lib/node_modules/npm
```

Comment (akoeplinger) [reply src/cbl-mariner/2.0/webassembly/amd64/Dockerfile:9]: I'd prefer not to touch 2.0 if we can avoid it.

Comment (akoeplinger) [reply src/cbl-mariner/2.0/webassembly/amd64/Dockerfile]: I've removed the change since the PR looks good otherwise.

**Review by akoeplinger — APPROVED** [review]

File: src/cbl-mariner/2.0/webassembly/amd64/Dockerfile (hunk 2 above)
Comment (am11) [reply src/cbl-mariner/2.0/webassembly/amd64/Dockerfile]: Alrighty then.I don't mind either way. Being explicit vs. relying on transitive dependencies. 👍

**Pull request merged**